Tests/tests/test_RDKShell.cpp

```diff
@@ -0,0 +1,1099 @@
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setScreenResolution")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setTopmost")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setVirtualResolution")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setVisibility")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setGraphicsFrameRate")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("showCursor")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("showFullScreenImage")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("showSplashLogo")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("showWatermark")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("suspend")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("suspendApplication")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("keyRepeatConfig")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("setAVBlocked")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getBlockedAVApplications")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("addEasterEggs")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("removeEasterEggs")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("enableEasterEggs")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("getEasterEggs")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("launchFactoryApp")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("launchFactoryAppShortcut")));
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("enableInputEvents")));
+    }
+TEST_F(RDKShellTest, enableInputEvents)
+{
+   ON_CALL(compositormock, enableInputEvents(::testing::_, ::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](const std::string& client, bool enable){
+                      EXPECT_EQ(client, string("searchanddiscovery"));
+                      EXPECT_EQ(enable, true);
+                      return false;
+                   }));
+
+   EXPECT_EQ(Core::ERROR_GENERAL, handler.Invoke(connection, _T("enableInputEvents"), _T("{"
+                                                                                "\"clients\": ["
+                                                                                "    \"searchanddiscovery\""
+                                                                                "],"
+                                                                                "\"enable\": true"
+                                                                                "}"), response));
+   ON_CALL(compositormock, enableInputEvents(::testing::_, ::testing::_))
+            .WillByDefault(::testing::Invoke(
+                [](const std::string& client, bool enable){
+		      EXPECT_EQ(client, string("searchanddiscovery"));
+		      EXPECT_EQ(enable, true);
+                      return true;
+                   }));
+
+   EXPECT_EQ(Core::ERROR_NONE, handler.Invoke(connection, _T("enableInputEvents"), _T("{"
```

Comment:
handler.Invoke doesn't need a previous handler.Invoke. please don't mix several tests in one. if it can be a separate test make it a separate test. this applies to all tests in this file